### PR TITLE
Custom rule updates

### PR DESCRIPTION
**Pull request by jamarw**
1. Separated zone-moniker rule into through different rules (syntax, zone and moniker).  Each extension will have it's own rule file going forward.  Updated rule names to follow "docs.{rule}" convention.
2. Added common.js to centralize regex and shared functions. 
3. Added checkMarkdownlintCustomProperty() to extension.ts.  This method is used to check for and preserve any markdownlint custom rules that a user has in addition to adding an entry for the docs custom rule path.
4. Deleted zone-moniker.js as it's no longer needed.

### CODE REVIEW

File: docs-markdown/markdownlint-custom-rules/common.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// @ts-check
+
+"use strict";
+
+// Triple colon
+module.exports.singleColon = /^:/gm;
+module.exports.tripleColonSyntax = /^:::/gm;
+module.exports.validTripleColon = /^:::\s+/gm;
```
Comment (nschonni) [inline docs-markdown/markdownlint-custom-rules/common.js]: Does space at the beginning of the `:::` make it invalid?

Comment (jamarw) [reply docs-markdown/markdownlint-custom-rules/common.js]: Hi @nschonni , currently the ::: should start at the beginning of a line with no whitespace.  Not sure if that may change going forward.  Thanks

Comment (nschonni) [reply docs-markdown/markdownlint-custom-rules/common.js]: 👍 should there be a test for the the triple colon with leading spaces as a failure then?

Comment (jamarw) [reply docs-markdown/markdownlint-custom-rules/common.js]: Good call, I'll let the PM know for follow-up.  Thanks

**Pull request merged**